cpp/roaring64map.hh

```diff
@@ -501,8 +501,8 @@ class Roaring64Map{
     size_t write(char *buf, bool portable = true) const {
         const char* orig = buf;
         // push map size
-        *((uint32_t*)buf) = roarings.size();
-        buf += sizeof(uint32_t);
+        *((uint64_t*)buf) = roarings.size();
```

Comment:
an unfortunate mistake, carnality being a 64-bit number for a 32-bit data structure strikes again!